replication failed when Deployment security is set `Notation`

1. push an image to harbor and sign it with notation
<img width="778" alt="Screenshot 2023-07-27 at 17 31 54" src="https://github.com/goharbor/harbor/assets/6709992/f3e78d83-c0fe-48b4-bd70-d92b8be40972">
2. set the `Deployment security` to be `Notation`
<img width="1020" alt="Screenshot 2023-07-27 at 17 32 17" src="https://github.com/goharbor/harbor/assets/6709992/12a3568a-d0d6-41a0-8606-e78a5221bc7b">

3. `docker pull 1.2.3.4/library/nginx:1.23.3` succeeded
However, when creating a pull-based replication on another harbor instance (same harbor version), it failed. Log:
```
2023-07-27T09:26:53Z [INFO] [/controller/replication/transfer/image/transfer.go:139]: client for source registry [type: harbor, URL: https://<harbor-ip>, insecure: true] created
2023-07-27T09:26:53Z [INFO] [/controller/replication/transfer/image/transfer.go:149]: client for destination registry [type: harbor, URL: http://core:8080, insecure: true] created
2023-07-27T09:26:53Z [INFO] [/controller/replication/transfer/image/transfer.go:182]: copying library/nginx:[1.23.3,sha256:6a90207e78cf72db3ea8a0461c0582ff33cc45d31c21c2c51ff18bacbfae4c01](source registry) to library/nginx:[1.23.3,sha256:6a90207e78cf72db3ea8a0461c0582ff33cc45d31c21c2c51ff18bacbfae4c01](destination registry)...
2023-07-27T09:26:53Z [INFO] [/controller/replication/transfer/image/transfer.go:210]: copying library/nginx:1.23.3(source registry) to library/nginx:1.23.3(destination registry)...
2023-07-27T09:26:53Z [INFO] [/controller/replication/transfer/image/transfer.go:467]: pulling the manifest of artifact library/nginx:1.23.3 ...
2023-07-27T09:26:53Z [INFO] [/controller/replication/transfer/image/transfer.go:473]: the manifest of artifact library/nginx:1.23.3 pulled
2023-07-27T09:26:53Z [INFO] [/controller/replication/transfer/image/transfer.go:226]: the artifact library/nginx:1.23.3 already exists on the destination registry, skip
2023-07-27T09:26:53Z [INFO] [/controller/replication/transfer/image/transfer.go:210]: copying library/nginx:sha256:6a90207e78cf72db3ea8a0461c0582ff33cc45d31c21c2c51ff18bacbfae4c01(source registry) to library/nginx:sha256:6a90207e78cf72db3ea8a0461c0582ff33cc45d31c21c2c51ff18bacbfae4c01(destination registry)...
2023-07-27T09:26:53Z [INFO] [/controller/replication/transfer/image/transfer.go:467]: pulling the manifest of artifact library/nginx:sha256:6a90207e78cf72db3ea8a0461c0582ff33cc45d31c21c2c51ff18bacbfae4c01 ...
2023-07-27T09:26:53Z [ERROR] [/controller/replication/transfer/image/transfer.go:470]: failed to pull the manifest of artifact library/nginx:sha256:6a90207e78cf72db3ea8a0461c0582ff33cc45d31c21c2c51ff18bacbfae4c01: http status code: 412, body: {"errors":[{"code":"PROJECTPOLICYVIOLATION","message":"The image is not signed by notation."}]}
2023-07-27T09:26:53Z [ERROR] [/controller/replication/transfer/image/transfer.go:194]: http status code: 412, body: {"errors":[{"code":"PROJECTPOLICYVIOLATION","message":"The image is not signed by notation."}]}
2023-07-27T09:26:53Z [ERROR] [/controller/replication/transfer/image/transfer.go:200]: got error during the whole transfer period, mark the job failure
```
after unselecting the `Notation` for `Deployment security`, the replication is successful.

4. for push-bashed replication, it is also succeeded without `Notation` for `Deployment security`. But failed when `Notation` is selected. Log:
```
2023-07-27T09:48:09Z [INFO] [/controller/replication/transfer/image/transfer.go:139]: client for source registry [type: harbor, URL: http://core:8080, insecure: true] created
2023-07-27T09:48:09Z [INFO] [/controller/replication/transfer/image/transfer.go:149]: client for destination registry [type: harbor, URL: https://<destination-harbor-ip>, insecure: true] created
2023-07-27T09:48:09Z [INFO] [/controller/replication/transfer/image/transfer.go:182]: copying project1/nginx:[1.23.3,sha256:6a90207e78cf72db3ea8a0461c0582ff33cc45d31c21c2c51ff18bacbfae4c01](source registry) to project1/nginx:[1.23.3,sha256:6a90207e78cf72db3ea8a0461c0582ff33cc45d31c21c2c51ff18bacbfae4c01](destination registry)...
2023-07-27T09:48:09Z [INFO] [/controller/replication/transfer/image/transfer.go:210]: copying project1/nginx:1.23.3(source registry) to project1/nginx:1.23.3(destination registry)...
2023-07-27T09:48:09Z [INFO] [/controller/replication/transfer/image/transfer.go:467]: pulling the manifest of artifact project1/nginx:1.23.3 ...
2023-07-27T09:48:09Z [INFO] [/controller/replication/transfer/image/transfer.go:473]: the manifest of artifact project1/nginx:1.23.3 pulled
2023-07-27T09:48:09Z [INFO] [/controller/replication/transfer/image/transfer.go:226]: the artifact project1/nginx:1.23.3 already exists on the destination registry, skip
2023-07-27T09:48:09Z [INFO] [/controller/replication/transfer/image/transfer.go:210]: copying project1/nginx:sha256:6a90207e78cf72db3ea8a0461c0582ff33cc45d31c21c2c51ff18bacbfae4c01(source registry) to project1/nginx:sha256:6a90207e78cf72db3ea8a0461c0582ff33cc45d31c21c2c51ff18bacbfae4c01(destination registry)...
2023-07-27T09:48:09Z [INFO] [/controller/replication/transfer/image/transfer.go:467]: pulling the manifest of artifact project1/nginx:sha256:6a90207e78cf72db3ea8a0461c0582ff33cc45d31c21c2c51ff18bacbfae4c01 ...
2023-07-27T09:48:09Z [ERROR] [/controller/replication/transfer/image/transfer.go:470]: failed to pull the manifest of artifact project1/nginx:sha256:6a90207e78cf72db3ea8a0461c0582ff33cc45d31c21c2c51ff18bacbfae4c01: http status code: 412, body: {"errors":[{"code":"PROJECTPOLICYVIOLATION","message":"The image is not signed by notation."}]}
2023-07-27T09:48:09Z [ERROR] [/controller/replication/transfer/image/transfer.go:194]: http status code: 412, body: {"errors":[{"code":"PROJECTPOLICYVIOLATION","message":"The image is not signed by notation."}]}
2023-07-27T09:48:09Z [ERROR] [/controller/replication/transfer/image/transfer.go:200]: got error during the whole transfer period, mark the job failure
```

--- a/src/server/middleware/util/util.go
+++ b/src/server/middleware/util/util.go
@@ -64,7 +64,7 @@ func SkipPolicyChecking(r *http.Request, projectID, artID int64) (bool, error) {
 
 	// 1, scanner pull access can bypass.
 	// 2, cosign/notation pull can bypass, it needs to pull the manifest before pushing the signature.
-	// 3, pull cosign signature can bypass.
+	// 3, pull cosign/notation signature can bypass.
 	if ok && secCtx.Name() == "v2token" {
 		if secCtx.Can(r.Context(), rbac.ActionScannerPull, project.NewNamespace(projectID).Resource(rbac.ResourceRepository)) ||
 			(secCtx.Can(r.Context(), rbac.ActionPush, project.NewNamespace(projectID).Resource(rbac.ResourceRepository)) &&
@@ -79,7 +79,7 @@ func SkipPolicyChecking(r *http.Request, projectID, artID int64) (bool, error) {
 	if err != nil {
 		return false, err
 	}
-	if len(accs) > 0 && accs[0].GetData().Type == model.TypeCosignSignature {
+	if len(accs) > 0 && (accs[0].GetData().Type == model.TypeCosignSignature || accs[0].GetData().Type == model.TypeNotationSignature) {
 		return true, nil
 	}
 

